test/sql/catalog/test_set_search_path.test

```diff
@@ -0,0 +1,110 @@
+# name: test/sql/catalog/test_set_search_path.test
+# description: SET schema and SET search_path
+# group: [catalog]
+
+# create a couple of schemas with a table each
+statement ok
+CREATE TABLE main_table(i INTEGER);
+
+statement ok
+CREATE SCHEMA test;
+
+statement ok
+CREATE TABLE test.test_table(i INTEGER);
+
+statement ok
+CREATE SCHEMA test2;
+
+statement ok
+CREATE TABLE test2.test2_table(i INTEGER);
+
+# Reading the config - It should be empty for now.
+statement error
+SELECT CURRENT_SETTING('search_path');
+
+statement error
+SELECT CURRENT_SETTING('schema');
+
+# Setting the default value.
+statement ok
+SET SEARCH_PATH = 'test';
+
+statement ok
+SET SEARCH_PATH = 'test,test2';
+
+statement error
+SET SEARCH_PATH = 'does_not_exist';
+
+# Setting the default value through 'schema'
+
+statement ok
+SET SCHEMA = 'test';
+
+statement error
+SET SCHEMA = 'test,test2';
```

Comment:
Should this give an error? Remember that schemas can contain commas (although this might be ill-advised!).

Reply:
Oh good point! Here is what I found:

 * PG supports quoting like`SET SEARCH_PATH '"foo","bar,baz."'` although it doesn't support escape. I added this too.
 * SET SCHEMA is deprecated and now gone in the latest PG. I'd leave the support in duckdb though as people seem like it.
 * Apparently our CREATE SCHEMA doesn't support comma-included names so I couldn't write  SQL test. The logic is covered by the unit tests so I hope it is sufficient.